sources/core/Game.cpp

```diff
@@ -707,15 +707,15 @@ void Game::generateUnsync()
     }
 }
 
-void Game::AddLastMessages(std::stringstream& stream)
+void Game::AddLastMessages(QByteArray& data)
 {
-    stream << std::endl;
+    data.append('\n');
     for (int i = (log_pos_ + 1) % messages_log_.size();
              i != log_pos_;
              i = (i + 1) % messages_log_.size())
     {
-        stream << messages_log_[i].type << " ";
-        stream << messages_log_[i].json.toStdString() << std::endl;
+        data.append(messages_log_[i].type + " ");
+        data.append(messages_log_[i].json + '\n');
```

Comment:
The dumps show that there is an error.
`" "` is actually a pointer, and `messages_log_[i].type` is a number, and you can do the pointer arithmetic in C++. So when you do that you get pointer in some memory area, and some random data is written into `data`.
It should be `QByteArray::number(messages_log_[i].type) + " "` instead.

If you fix that then remake the dumps test.